### PR TITLE
Refactor io

### PR DESCRIPTION
**Pull request by jfcorbett**
Merged "readers" and "writers" into a single package called "io".  (And corresponding tests similarly.)
This makes it easier to group things that belong together. e.g. can collect openpyxl-specific things in a single module.
It also shortens the API. 

Moved JSON things to this new io package.

Also tidied up the __init__.py files.

### CODE REVIEW

File: pdtable/io/_json.py [hunk 1]
```diff
@@ -56,6 +56,7 @@ def to_json_serializable(obj: JsonDataPrecursor) -> JsonData:
     if isinstance(obj, np.ndarray):
         if f"{obj.dtype}" == "float64":
             return [val if (not np.isnan(val)) else None for val in obj.tolist()]
+            # TODO  Will fail if obj.ndim > 1. Should recurse here as well.
```
Comment (thpetsen) [inline pdtable/io/_json.py:59]: the method is not exposed externally, i.e. it only handles Startable-like objects, i.e. single dimension arrays

File: pdtable/io/parsers/blocks.py [hunk 2]
```diff
@@ -109,6 +109,7 @@ def make_table_json_precursor(cells: CellGrid, **kwargs) -> JsonDataPrecursor:
                 f"Unable to parse value in column '{name}' of table '{table_name}' as '{unit}'"
             ) from e
 
+    # TODO fixer should be responsible for reporting and raising errors. Parser shouldn't have to query fixer.
```
Comment (thpetsen) [inline pdtable/io/parsers/blocks.py:112]: It is a standard parser design that the parser should resolve as many errors as possible before stopping.
The fixer does not know when it "is done"
Only way that the fixer itself can stop_on_errors is by stopping at the first error it sees, which breaks any reporting &c.

File: pdtable/__init__.py [hunk 3]
```diff
@@ -6,13 +6,10 @@
```
Comment (thpetsen) [inline pdtable/__init__.py:6]: why is parse_blocks no longer included ? I miss it already !

File: pdtable/test/io/test_read_csv_fixer.py [hunk 4]
```diff
@@ -4,9 +4,9 @@
```
Comment (thpetsen) [inline pdtable/test/io/test_read_csv_fixer.py:4]: @jfcorbett you mention som "real-use" cases, where it is important that "" (empty cell is not found in float columns), cases like this should be included in test(s)

**Review by thpetsen — APPROVED** [review]

File: pdtable/io/_json.py (hunk 1 above)
Comment (jfcorbett) [reply pdtable/io/_json.py:59]: True.

File: pdtable/__init__.py (hunk 3 above)
Comment (jfcorbett) [reply pdtable/__init__.py:6]: What's the use case? 
I was thinking, typical usage of the pdtable package wouldn't really involve `parse_blocks()`. So it shouldn't be among the select few functions exposed at the top level. How about at the `pdtable.io` level?

`from pdtable.io import parse_blocks`

File: pdtable/test/io/test_read_csv_fixer.py (hunk 4 above)
Comment (jfcorbett) [reply pdtable/test/io/test_read_csv_fixer.py:4]: Yes, absolutely. I haven't gotten around to that.

File: pdtable/__init__.py (hunk 3 above)
Comment (thpetsen) [reply pdtable/__init__.py:6]: Agree, cellgrid bases methods need not be exposed at top level

Comment (thpetsen) [reply pdtable/__init__.py:6]: get_table_data should be exposed here since it is used w. read_bundle_from_csv() !?

Comment (jfcorbett) [reply pdtable/__init__.py:6]: TBH, I'm not super clear on the purpose of `read_bundle_from_csv()` is. It seems out of sync with the rest of the io API. 

Right now I'm inclined to remove it from the `__init__.py` until we figure out what to do with it.

File: pdtable/io/parsers/blocks.py (hunk 2 above)
Comment (jfcorbett) [reply pdtable/io/parsers/blocks.py:112]: We just agreed that lines 112-121 can be put in a fixer.report() method or sth like that.

**Pull request merged**